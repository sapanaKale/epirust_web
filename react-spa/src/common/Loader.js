import React from 'react';

export default function Loader() {
    return (
        <div id="loader" data-testid="loader" className="multi-spinners">
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
            <div className="spinner-grow text-dark" />
        </div>
    );
}