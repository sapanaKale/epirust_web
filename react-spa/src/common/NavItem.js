/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import {Link, useRouteMatch} from "react-router-dom";
import React from "react";

export function NavItem({ name, linksTo, activeOnExactMatch = false }) {
  const match = useRouteMatch({
    path: linksTo,
    exact: activeOnExactMatch
  });

  return (
    <li className={`nav-item`}>
      <Link className={`nav-link ${match ? "active" : ""}`} to={linksTo}>{name}</Link>
    </li>
  );
}