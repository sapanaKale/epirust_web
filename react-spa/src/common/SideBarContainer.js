/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React, {useState} from "react";
import {faTimes} from "@fortawesome/free-solid-svg-icons";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

export default function SideBarContainer({children}) {
  const [sidebarOpen, updateSideBarOpen] = useState(false);

  return (
    <div className="drawer-wrap">
      <button onClick={() => updateSideBarOpen(status => !status)} type="button" className="btn btn-primary" aria-label="Close">
        <span>Settings</span>
        <svg className="bi bi-gear-fill" width="1em" height="1em" viewBox="0 0 16 16" fill="currentColor"
             xmlns="http://www.w3.org/2000/svg">
          <path fillRule="evenodd"
                d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 01-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 01.872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 012.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 012.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 01.872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 01-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 01-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 100-5.86 2.929 2.929 0 000 5.858z"
                clipRule="evenodd"/>
        </svg>
      </button>
      <div className={`drawer ${sidebarOpen && "open"}`}>
        <FontAwesomeIcon onClick={() => updateSideBarOpen(status => !status)} icon={faTimes}/>
        {children}
      </div>
    </div>
  );
}
