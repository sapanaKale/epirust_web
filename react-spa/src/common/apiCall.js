import config from "../config";

export const post = (url, data) => {
  return fetch(`${config.API_HOST}/api${url}`, {
    method: 'POST',
    headers: {'Content-Type': 'application/json'},
    body: JSON.stringify(data)
  })
    .then(response => handleNon200(response))
}

export const get = (url) => {
  return fetch(`${config.API_HOST}/api${url}`, {
    method: 'GET',
    headers: {'Content-Type': 'application/json'}
  })
    .then(response => handleNon200(response))
}

const handleNon200 = (response) => {
  if (!response.ok) {
    throw Error(response.statusText);
  }
  return response;
}