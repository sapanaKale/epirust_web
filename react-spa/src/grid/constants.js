export const BaseColors = {
    YELLOW: "#fcce44",
    LIGHT_YELLOW: "#fff0c2",
    PURPLE: "#3498db",
    BLUE: "#007bff",
    ORANGE: "#ffa500",
    RED: "#e74c3c",
    GRAY_LIGHTER: "#f1f1f1",
    PINK: "#e83e8c",
    LIGHT_PINK: "#f9d1e4",
    GRAY: "#ccc",
    BLACK: "black",
    GREEN: "#28a745",
    WHITE: "#fff",
    LIGHT_ORANGE: "#fad5b5",
    LIGHT_BLUE: "#b9dafd"
};

export const AreaColors = {
    HOUSING: BaseColors.LIGHT_YELLOW,
    WORK: BaseColors.LIGHT_PINK,
    TRANSPORT: BaseColors.LIGHT_ORANGE,
    HOSPITAL: BaseColors.LIGHT_BLUE,
    OTHER: BaseColors.GRAY
};

export const LandmarkColors = {
    HOUSES: BaseColors.GRAY,
    OFFICES: BaseColors.GRAY
};

export const AgentStateToColor = {
    's': BaseColors.BLUE,
    'e': BaseColors.ORANGE,
    'i': BaseColors.RED,
    'r': BaseColors.GREEN,
    'd': BaseColors.BLACK
};

export const AgentStateMapper = {
    's': 'susceptible',
    'e': 'exposed',
    'i': 'infected',
    'r': 'recovered',
    'd': 'deceased'
};

export const Interventions = Object.freeze({
    LOCKDOWN: 'lockdown',
    BUILD_NEW_HOSPITAL: 'build_new_hospital',
    VACCINATION: 'vaccination',
    status: {
        LOCKDOWN_START: 'locked_down',
    }
});
