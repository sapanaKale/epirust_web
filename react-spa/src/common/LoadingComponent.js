/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import {LOADING_STATES} from "./constants";
import Loader from "./Loader";
import React from "react";
import PropTypes from 'prop-types'

export const LoadingComponent = (props) => {
  switch (props.loadingState) {
    case LOADING_STATES.FINISHED:
      return props.children;

    case LOADING_STATES.FAILED:
      return <div>Failed to load</div>

    default:
      return <Loader/>
  }
}

LoadingComponent.propTypes = {
  loadingState: PropTypes.oneOf([1, 2, 3]).isRequired
}