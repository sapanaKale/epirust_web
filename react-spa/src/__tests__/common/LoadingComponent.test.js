/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
import {LoadingComponent} from "../../common/LoadingComponent";
import {LOADING_STATES} from "../../common/constants";
import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';


describe('LoadingComponent', () => {
  it('should render loader if the state is loading', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<LoadingComponent loadingState={LOADING_STATES.LOADING}>
      <div>Test child component</div>
    </LoadingComponent>)

    expect(renderer.getRenderOutput()).toMatchSnapshot()
  });

  it('should render error message if the state is failed', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<LoadingComponent loadingState={LOADING_STATES.FAILED}>
      <div>Test child component</div>
    </LoadingComponent>)

    expect(renderer.getRenderOutput()).toMatchSnapshot()
  });

  it('should render child component if the state is finished', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<LoadingComponent loadingState={LOADING_STATES.FINISHED}>
      <div>Test child component</div>
    </LoadingComponent>)

    expect(renderer.getRenderOutput()).toMatchSnapshot()
  });
});
