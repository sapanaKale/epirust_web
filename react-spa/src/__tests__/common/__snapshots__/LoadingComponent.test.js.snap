// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingComponent should render child component if the state is finished 1`] = `
<div>
  Test child component
</div>
`;

exports[`LoadingComponent should render error message if the state is failed 1`] = `
<div>
  Failed to load
</div>
`;

exports[`LoadingComponent should render loader if the state is loading 1`] = `<Loader />`;
